crates/swc/src/lib.rs

```diff
@@ -269,99 +269,119 @@ impl Compiler {
+                                 string",
+                            )?))
+                        }
+                        None => {
+                            bail!("failed to parse inline source map: `sourceMappingURL` not found")
+                        }
+                    }
+                };
+
             // Load original source map
             match input_src_map {
                 InputSourceMap::Bool(false) => Ok(None),
                 InputSourceMap::Bool(true) => {
                     let s = "sourceMappingURL=";
                     let idx = fm.src.rfind(s);
-                    let src_mapping_url = idx.map(|idx| &fm.src[idx + s.len()..]);
+                    let data_url = idx.map(|idx| &fm.src[idx + s.len()..]);
+
+                    match read_inline_sourcemap(data_url) {
+                        Ok(r) => Ok(r),
+                        Err(_) => {
+                            // Load original source map if possible
+                            match &name {
+                                FileName::Real(filename) => {
+                                    let dir = match filename.parent() {
+                                        Some(v) => v,
+                                        None => {
+                                            bail!(
+                                                "unexpected: root directory is given as a input \
+                                                 file"
+                                            )
+                                        }
+                                    };
 
-                    // Load original source map if possible
-                    match &name {
-                        FileName::Real(filename) => {
-                            let dir = match filename.parent() {
-                                Some(v) => v,
-                                None => {
-                                    bail!("unexpected: root directory is given as a input file")
-                                }
-                            };
+                                    let path = match data_url {
+                                        Some(data_url) => dir.join(data_url).display().to_string(),
+                                        None => {
+                                            format!("{}.map", dir.join(filename).display())
+                                        }
+                                    };
 
-                            let path = match src_mapping_url {
-                                Some(src_mapping_url) => {
-                                    dir.join(src_mapping_url).display().to_string()
-                                }
-                                None => {
-                                    format!("{}.map", dir.join(filename).display())
-                                }
-                            };
+                                    let file = File::open(&path)
```

Comment:
Please use `read_to_string` instead.

Reply:
@kdy1 sourcemap::SourceMap::from_reader(file)

Reply:
Oh I looked at source code and it uses BufReader internally.